### PR TITLE
Disable S3 filepicker when no `ui.sourceBuckets` and set default source bucket

### PR DESCRIPTION
**Pull request by fiskus**
Screenshot from 2021-05-29 12-13-26

Spec: https://github.com/quiltdata/specs/pull/16

### CODE REVIEW

File: catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx
Comment (nl0) [inline catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx:1327]: `bucket` is always a string (non-nullable), so what's the point checking `!!bucket`?

File: catalog/app/constants/urls.js
Comment (nl0) [inline catalog/app/constants/urls.js:11]: why do we need a different url here?

**Review by nl0 — COMMENTED** [review]
looks good, but we should decide on setting the default bucket and incorporate that logic into code populating the source buckets

File: catalog/app/constants/urls.js
Comment (fiskus) [reply catalog/app/constants/urls.js:11]: There is no https://docs.quiltdata.com/catalog/preferences but https://docs.quiltdata.com/v/master/catalog/preferences

When we deploy a new version (v.3.5) first link will become available https://docs.quiltdata.com/catalog/preferences. And we could find all `docs35` and replace them with `docs` (Catalog Preferences, Athena Queries etc.)

Comment (nl0) [reply catalog/app/constants/urls.js:11]: maybe call the variable `docsMaster` then?

Comment (fiskus) [reply catalog/app/constants/urls.js:11]: I don't mind to use `docsMaster`.
My reason was that `docsMaster` is a good name and you don't have the itch to replace it with just `docs`.

**Comment by fiskus** [comment]
Why can we merge it separately from the default buckets feature? This PR hides S3FilePicker if no source buckets. If a user wants to populate a package with the same bucket he needs to set `sourceBuckets` field in config.

**Comment by nl0** [comment]
> Why can we merge it separately from the default buckets feature? This PR hides S3FilePicker if no source buckets. If a user wants to populate a package with the same bucket he needs to set `sourceBuckets` field in config.

well, we need at least some way to predictably set the default bucket. prior to this PR it was always set to the current bucket, after this PR is merged the default bucket is unpredictable (if there's more than 1 in the config)

**Review by nl0 — APPROVED** [review]

**Pull request merged**
